### DISCUSSION TITLE
dockerless install update

### DISCUSSION THREAD

**Discussion by sysadminpower2019** [dani-garcia/vaultwarden#4397] [General]
Hello all,

i installed vaultwarden without docker quite some time ago. It has been working perfectly but i have realized that the version I have on my system is very outdated and possibly dangerous.
Apparently the original guide I used for the install is also no longer available. 
What would be the process to update this install? I suppose that only the data in the "data" folder needs to be preserved if not using an external db?

**Comment by BlackDex** [comment]
In that case all in the data folder is the important stuff.
Regarding on how to update. I have no clue. There are several community supported dockerless install options.

See https://github.com/dani-garcia/vaultwarden/wiki/Third-party-packages

**Comment by sysadminpower2019** [comment]
should it be enough to simply replace all the files in the /var/lib/vaultwarden besides the data folder with the new data?
/var/lib/vaultwarden seems to be where everything is installed

**Reply by stefan0xC** [reply]
We don't know how you set up your system or what's in `/var/lib/vaultwarden` (besides the `data` folder), so you did not provide enough information for us to answer your questions.

To update Vaultwarden manually you need to replace both the `vaultwarden` binary and if you are using the web vault, you should also replace the contents of the `web-vault` folder (or point `WEB_VAULT_FOLDER` to the updated web-vault).

I'd recommend reading the [configuration overview](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview) (and related pages) if you can't remember how you setup your system. Some things you have to figure out yourself (i.e. if you use systemd to start vaultwarden or if you use a different init system, etc).

**Comment by the-marat** [comment]
Here is mine script for update vaultwarden without docker 

just replace with you version

version1
cd /var/lib/vaultwarden
wget https://github.com/dani-garcia/bw_web_builds/releases/download/v2024.5.1b/bw_web_v2024.5.1b.tar.gz
tar -xvf bw_web_v2024.5.1b.tar.gz
rm bw_web_v2024.5.1b.tar.gz
chmod +x /usr/local/bin/vaultwarden
chown -R vaultwarden:vaultwarden /var/lib/vaultwarden
cd ~/source
delete vaultwarden folder!!!! - ~/source/vaultwarden
git clone https://github.com/dani-garcia/vaultwarden.git # clones the repo
cd vaultwarden
cargo build --features sqlite --release
systemctl stop vaultwarden
cp ~/source/vaultwarden/target/release/vaultwarden /usr/local/bin/vaultwarden
chmod +x /usr/local/bin/vaultwarden
chown -R vaultwarden:vaultwarden /var/lib/vaultwarden
systemctl start vaultwarden


version2
systemctl stop vaultwarden
cd ~/source
git clone https://github.com/dani-garcia/vaultwarden.git
cd vaultwarden
cargo build --features sqlite --release
cd /var/lib/vaultwarden
cp ~/source/vaultwarden/target/release/vaultwarden /usr/local/bin/vaultwarden
systemctl stop vaultwardenchmod +x /usr/local/bin/vaultwarden
chmod +x /usr/local/bin/vaultwarden
useradd -m -d /var/lib/vaultwarden vaultwarden
chown -R vaultwarden:vaultwarden /var/lib/vaultwarden
cd /var/lib/vaultwarden
wget https://github.com/dani-garcia/bw_web_builds/releases/download/v2024.6.2c/bw_web_v2024.6.2c.tar.gz
tar -xvf bw_web_v2024.6.2c.tar.gz
rm bw_web_v2024.6.2c.tar.gz
systemctl start vaultwarden

**Comment by Anuril** [comment]
Bit late to the party - it sounds like you've used https://gist.github.com/heinoldenhuis/f8164f73e5bff048e76fb4fff2e824e1 (not available anymore). by @heinoldenhuis 

Based on that gist, I've build an install / upgrade script: https://github.com/Anuril/vwinst 

It currently works on debian 11 + 12, results with rhel based distros might differ, and it might not work for your instance.

**Reply by BlackDex** [reply]
An archived version can be found here: https://web.archive.org/web/20231023085442/https://gist.github.com/heinoldenhuis/f8164f73e5bff048e76fb4fff2e824e1

**Comment by svn-josh** [comment]
Here is a Script that automates the Docker less Update:

https://github.com/svn-josh/vaultwarden_autoupdate.git
